### PR TITLE
Change packages installed in go-agent.

### PR DESCRIPTION
**Pull request by doctoryes**
@feanil @macdiesel Removed tubular. Added Docker and AWS CLI.

For your review.

### CODE REVIEW

File: docker/build/go-agent/Dockerfile [hunk 1]
```diff
@@ -18,17 +18,20 @@ WORKDIR /data
...
     python-pip
 
-# TODO: repalce this with a pip install command so we can version this properly
-RUN git clone https://github.com/edx/tubular.git /opt/tubular
-RUN pip install -r /opt/tubular/requirements.txt
-RUN cd /opt/tubular;python setup.py install
+# Install Docker - for Docker container building by a go-agent.
+RUN \
+  apt-get install -y curl && \
+  curl -fsSL https://get.docker.com/ | sh
```
Comment (feanil) [inline docker/build/go-agent/Dockerfile]: Curling to bash makes me a bit wary?  Can we just do the exact things we need instead?

```
            $sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D"
            $sh_c "mkdir -p /etc/apt/sources.list.d"
            $sh_c "echo deb [arch=$(dpkg --print-architecture)] https://apt.dockerproject.org/repo ${lsb_dist}-${dist_version} ${repo} > /etc/apt/sources.list.d/docker.list"
            $sh_c 'sleep 3; apt-get update; apt-get install -y -q docker-engine'
```

I think this is all we need:
- Add the repo
- Install docker-engine

Comment (doctoryes) [inline docker/build/go-agent/Dockerfile]: `curl -L http://mysketchysite.com/weird_script | sh`? What could go wrong? 😄 

How about instead I commit the Docker install script and use it during installation? That way, we don't have to figure out the important parts - and updating it is as simple as downloading a new version in the future.

Comment (feanil) [inline docker/build/go-agent/Dockerfile]: That sounds good to me.

**Comment by doctoryes** [comment]
@macdiesel @feanil This PR removes tubular from the go-agent Docker container and adds a couple of other things. Please review.

**Comment by macdiesel** [comment]
👍

**Pull request merged**